spring-boot-project/spring-boot/src/main/java/org/springframework/boot/ansi/Ansi256PropertySource.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.ansi;
+
+import org.springframework.core.env.PropertyResolver;
+import org.springframework.core.env.PropertySource;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link PropertyResolver} for {@link Ansi256Color.Background} and
+ * {@link Ansi256Color.Foreground} elements. Supports properties of the form
+ * {@code Ansi256Color.Foreground_N} and {@code Ansi256Color.Background_N} ({@code N} must
+ * be between 0 and 255).
+ *
+ * @author Toshiaki Maki
+ * @since 2.2.0
+ */
+public class Ansi256PropertySource extends PropertySource<AnsiElement> {
```

Comment:
I wonder if it would be better to roll this into the existing `AnsiPropertySource` class?